react-state/my-app/src/ToggleButton.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'react';
+
+type Props = {
+  text: string;
+  color: string;
+};
+
+export function ToggleButton({ text, color }: Props) {
+  const [isClicked, setisClicked] = useState(false);
+  console.log('useState', isClicked);
+
+  function handleClick() {
+    console.log('before setter', isClicked);
+    setisClicked(!isClicked);
+    console.log('after setter', isClicked);
+  }
+
+  if (isClicked) {
+    return (
+      <button
+        onClick={handleClick}
+        style={{
+          backgroundColor: color,
+        }}>
+        {text}
+      </button>
+    );
+  }
+
+  return (
+    <button
+      onClick={handleClick}
+      style={{
+        backgroundColor: 'white',
+      }}>
+      {text}
+    </button>
+  );
+}
```

Comment:
Good solution, but the only difference between the two return statements is the background color. Can you figure out how to get rid of all that code duplication?